jenkins/docker/Jenkinsfile

```diff
@@ -41,25 +53,77 @@ pipeline {
+                        }
+                        def CREDENTIAL_ID = "jenkins-staging-docker-prod-token"
                         sh "echo Switch to Production"
+                        withCredentials([usernamePassword(credentialsId: CREDENTIAL_ID, usernameVariable: 'DOCKER_USERNAME', passwordVariable: 'DOCKER_PASSWORD')]) {
+                            sh '''
+                                set -ex
+                                echo Login to $CREDENTIAL_ID
+                                curl -L https://github.com/google/go-containerregistry/releases/latest/download/go-containerregistry_Linux_x86_64.tar.gz \\
+                                    -o go-containerregistry.tar.gz
+                                tar -zxvf go-containerregistry.tar.gz
+                                chmod +x gcrane
+                                mv gcrane /usr/local/bin/
+                                docker logout && docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
+                                gcrane cp ${SOURCE_IMAGE_IN_STAGING} ${DESTINATION_IMAGE_IN_PROD}
+                            '''
+                        }
+                    } else {
+                        def CREDENTIAL_ID = "jenkins-staging-docker-staging-credential"
+                        withCredentials([usernamePassword(credentialsId: CREDENTIAL_ID, usernameVariable: 'DOCKER_USERNAME', passwordVariable: 'DOCKER_PASSWORD')]) {
+                            sh '''
+                                set -ex
+                                echo Login to $CREDENTIAL_ID
+                                docker logout && docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD && eval $DOCKER_BUILD_SCRIPT_WITH_COMMANDS
+                            '''
+                        }
+                    }
+                }
+            }
+            post() {
+                always {
+                    script {
+                        postCleanup()
+                        sh "docker logout && docker image prune -f --all"
                     }
-                    sh "echo Account: $CREDENTIAL_ID"
-                    withCredentials([usernamePassword(credentialsId: CREDENTIAL_ID, usernameVariable: 'DOCKER_USERNAME', passwordVariable: 'DOCKER_PASSWORD')]) {
+                }
+            }
+        }
+        stage('docker-build-ecr') {
+            agent {
+                docker {
+                    label 'Jenkins-Agent-Ubuntu2004-X64-m52xlarge-Docker-Builder'
+                    image 'opensearchstaging/ci-runner:ubuntu2004-x64-docker-buildx0.6.3-qemu5.0-awscli1.22-jdk14'
+                    args '-u root -v /var/run/docker.sock:/var/run/docker.sock'
+                    alwaysPull true
+                }
+            }
+            steps {
+                script {
+                    withAWS(role: 'abhng-ecr-role', roleAccount: "${ABHNG_ACCOUNT}", duration: 900, roleSessionName: 'jenkins-session') {
```

Comment:
Nice role and role name :D

Reply:
lol. That's just for testing. I'll update it when it's ready for review.